File: app/src/main/java/com/example/scoutquest/data/models/Game.kt
package com.example.scoutquest.data.models

import android.os.Parcelable
import com.google.firebase.firestore.DocumentId
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.RawValue

@Parcelize
data class Game(
    @DocumentId
    val gameId: String = "",
    val creatorId: String = "",
    val creatorEmail: String = "",
    val name: String = "",
    val description: String = "",
    val tasks: List<Task> = emptyList(),
    val isPublic: Boolean = false,

    //to implement in future:
    val playCount: Int? = null,
    val rating:@RawValue Rating? = null
): Parcelable

File: models/game.py
from pydantic import BaseModel
from typing import List, Optional
from .task import Task, TaskCreate

class GameBase(BaseModel):
    name: str
    description: str
    isPublic: bool

class GameCreate(GameBase):
    tasks: List[TaskCreate]

class Game(GameBase):
    gameId: int
    creator: str
    tasks: List[Task]

    class Config:
        orm_mode = True
